clubs/permissions.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import permissions
+from clubs.models import Membership
+
+
+class IsSuperuser(permissions.BasePermission):
+    """
+    Grants permission if the current user is a superuser.
+    """
+    def has_object_permission(self, request, view, obj):
+        return request.user.is_authenticated and request.user.is_superuser
+
+    def has_permission(self, request, view):
+        return request.user.is_authenticated and request.user.is_superuser
+
+
+class ClubPermission(permissions.BasePermission):
+    """
+    Only club owners should be able to delete.
+    Club owners and officers should be able to update.
+    Anyone with permission should be able to create.
+    """
+    def has_object_permission(self, request, view, obj):
+        if view.action in ['retrieve']:
```

Comment:
In general for permissions it seems DRF recommends checking `request.method` instead of `view.action`, not sure if it makes a difference, but I think that way is slightly clearer to understand ([docs](https://www.django-rest-framework.org/api-guide/permissions/#custom-permissions)).

Reply:
In the cases mentioned in the documentation, it looks like there's no example with a `ViewSet`. In those cases, it feels appropriate to use `request.method`. My rationale was that even if the method type changes (we switch from a PUT to a POST or a PATCH to a PUT), the permissions system still works correctly. Let me know what you think!